File: 5-longest-palindromic-substring/longest-palindromic-substring.java
class Solution {
   public String longestPalindrome(String s) {
	
    int len = 0;
    String ans = "";
    int l = 0;
    int r = 0;

    for (int i = 0 ; i < s.length();i++){
         l = i;
         r = i;
        while(l >= 0 && r < s.length() && s.charAt(l) == s.charAt(r)){
            if (len < r-l+1){
            ans = s.substring(l,r+1);
            len = r - l+1;
        }
            l--;
            r++;

        }

        

    }

    for (int i = 0 ; i < s.length();i++){
         l = i;
         r = i+1;
        while(l >= 0 && r < s.length() && s.charAt(l) == s.charAt(r)){
            if (len < r-l+1){
            ans = s.substring(l,r+1);
            len = r - l+1;
        }
            l--;
            r++;

        }

        

    }

    return ans;
}

}

File: 5-longest-palindromic-substring/longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        ans = ""
        for mid in range(len(s)):

            start = mid
            end = mid 
            
            while start >= 0 and end < len(s) and s[start] == s[end]:
                if len(ans) < end - start + 1:
                    ans = s[start:end + 1]
                start -= 1
                end += 1
            
        for mid in range(len(s)-1):

            start = mid
            end = mid + 1
            
            while start >= 0 and end < len(s) and s[start] == s[end]:
                if len(ans) < end - start + 1:
                    ans = s[start:end + 1]
                start -= 1
                end += 1
        return ans


                
